### PR TITLE
[WFLY-9893] add tests for script wsprovide and wsconsume

### PR DESCRIPTION
**Pull request by rsearls**
Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:
- [ ] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue(s)
- [ ] Pull Request contains description of the issue(s)
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [ ] Pull Request requires a change to the documentation
- [ ] Documentation have been updated accordingly
- [ ] Tests were added to cover changes

For new features ensure as well:
- [ ] Analysis was done
- [ ] Test Plan has been done
- [ ] Tests were verified in advance

If you are not an active contributor of the WildFly project you can request sponsorship by one of the members to help guide you through the process.

### CODE REVIEW

**Comment by rsearls** [comment]
https://issues.jboss.org/browse/WFLY-9893

**Comment by kabir** [comment]
@rsearls compilation error

**Comment by rsearls** [comment]
Resolved

**Comment by stuartwdouglas** [comment]
The tests are failing

File: testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/scripts/WSProvideScriptTestCase.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2012, Red Hat Middleware LLC, and individual contributors
```
Comment (rhusar) [inline testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/scripts/WSProvideScriptTestCase.java]: My watch says its 2018 already

**Review by rhusar — CHANGES_REQUESTED** [review]
Are those binary files (.class) intentionally committed? That doesn't seem right.

**Comment by rsearls** [comment]
Yes the class file is intended and required required.  It is the input to wsprovide which generates jax-ws artifacts based upon the input class.

**Comment by rhusar** [comment]
But why wouldn't the class files be compiled during compilation of the integration tests?

File: testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/scripts/ScriptUtil.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+     * @param env environment
+     * @throws IOException if I/O error occurs
+     */
+    private static String executeCommand(List<String> command, OutputStream os,
+                                         String message, Map<String, String> env) throws IOException, InterruptedException {
+
+        ProcessBuilder pb = new ProcessBuilder(command);
+        if (env != null) {
+            for (String variable : env.keySet()) {
+                pb.environment().put(variable, env.get(variable));
```
Comment (dmlloyd) [inline testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/scripts/ScriptUtil.java]: Regarding the test failures - maybe the M2_HOME environment variable isn't getting to the child process?  Or, maybe there are multiple (or no) Maven settings files on the server?  @ctomc do you know how we're presently configuring Maven settings on the server?

Comment (dmlloyd) [reply testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/scripts/ScriptUtil.java]: It's essentially failing to find xalan:serializer:2.7.1.jbossorg-4, which I can see exists at https://repository.jboss.org/nexus/service/local/repositories/thirdparty-releases/content/xalan/serializer/2.7.1.jbossorg-4/serializer-2.7.1.jbossorg-4.jar - hence why I think it's some kind of Maven environment setup issue.

JBoss Modules searches the following places for settings, in order:

* The URL found in the ``jboss.modules.settings.xml.url`` system property
* The ``.m2/settings.xml`` file inside of the user home directory (from the system property ``user.home``)
* The ``conf/settings.xml`` file inside of the directory in the ``M2_HOME`` environment variable

If that fails, then a set of default settings is used - which would exclude ``repository.jboss.org``.

Comment (ctomc) [reply testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/scripts/ScriptUtil.java]: we do use maven settings property everywhere in testsuite.
see https://github.com/wildfly/wildfly/blob/master/testsuite/integration/pom.xml#L275
this are the settings you are looking for.
or in short, you need to pass "maven.repo.local" system property.
Value you usually get from maven directly, ${settings.localRepository}

so just make sure that maven.repo.local is properly forwarded.

**Review by ctomc — CHANGES_REQUESTED** [review]
This whole test looks like a big mess.
Please clean it up, to not use any File API but Path one instead as looking all this concatenating and checking / or \ everywhere just calls for trouble.

Another thing I don't really like is that this test is introducing a new set of tests which is basically testing our startup scripts. And tests like this do not belong to WS testsuite or integration testsuite, but smoke tests maybe. In any case there shouldn't be any arquillian involved in it.

**Comment by rsearls** [comment]
Withdrawing PR for the time being.

**Pull request closed without merging**